pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java

```diff
@@ -290,6 +295,22 @@ public static FlattenStep newFlattenFunction(JsonNode node) {
     return builder.build();
   }
 
+  private static TransformStep newComputeFieldFunction(JsonNode node) {
+    List<ComputeField> fieldList = new ArrayList<>();
+    node.get("fields")
+        .iterator()
+        .forEachRemaining(
+            it ->
+                fieldList.add(
+                    ComputeField.builder()
+                        .name(it.get("name").asText())
+                        .expression(it.get("expression").asText())
+                        .type(ComputeFieldType.valueOf(it.get("type").asText()))
```

Comment:
we must validate  that the "type" is a supported type, this will prevent runtime errors.
also, the error should report the list of available times

Reply:
I think it's validated by the Json Schema. And the list of possible is in the error message. @aymkhalil do you confirm ?

Reply:
See https://github.com/datastax/pulsar-transformations/pull/19/files#diff-02d4ba68ec7c9d19097b2324e4a85fdf66c1f1d9266a976fa3e905d2f1f1ba01R129

Reply:
Confirmed. All validation logic we get out of the box from the JSON schema (and tested in unit tests). Thanks @cbornet for switching to schema based config, it takes away the validation burden!